pkg/matcher/cel.go

```diff
@@ -24,28 +24,93 @@ const (
 	changedFilesTags = "files."
 )
 
+// evaluateBranchExpression function evaluates the branch expression for the given branch key from the provided expression.
+// Examples of expressions:
+//   - target_branch == "refs/heads*"
+//   - source_branch == "refs/heads*"
+//
+// If source_branch and target_branch contain anything other than "refs/heads*", the function won't perform any action.
+func evaluateBranchExpression(expression, branchKey string) bool {
+	env, err := cel.NewEnv(cel.Declarations(
+		decls.NewVar(branchKey, decls.String),
+	))
+	if err != nil {
+		return false
+	}
+
+	// Define data map with placeholder values.
+	data := map[string]interface{}{
+		branchKey: "refs/heads/*",
+	}
+
+	out, err := evaluateCELExpression(env, expression, data)
+	if err != nil {
+		return false
+	}
+	// Return evaluation result as bool.
+	if boolVal, ok := out.Value().(bool); ok {
+		return boolVal
+	}
+	return false
+}
+
+func handleBranchCondition(event *info.Event, data map[string]interface{}, splitExprData, branchKey, branchFromEvent string) {
+	// For push event the target_branch & source_branch info coming from payload have refs/heads/
+	// but user may or mayn't provide refs/heads/ info while giving target_branch or source_branch in CEL expression
+	// ex:  pipelinesascode.tekton.dev/on-cel-expression: |
+	//        event == "push" && target_branch == "main" && "frontend/***".pathChanged()
+	// This logic will handle such case.
+	if event.TriggerTarget == "push" {
+		if !strings.Contains(splitExprData, "refs/heads/") {
```

Comment:
so we are not supporting if someone provides `refs/heads/main` in CEL expression?